e2e/logical_volume_test.go

```diff
@@ -0,0 +1,190 @@
+package e2e
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	"github.com/onsi/ginkgo/v2/types"
+	. "github.com/onsi/gomega"
+	topolvmlegacyv1 "github.com/topolvm/topolvm/api/legacy/v1"
+	topolvmv1 "github.com/topolvm/topolvm/api/v1"
+	clientwrapper "github.com/topolvm/topolvm/client"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const nsLogicalVolumeTest = "logical-volume"
+
+//go:embed testdata/logical_volume/pvc-template.yaml
+var pvcTemplateYAMLForLV string
+
+func testLogicalVolume() {
+	var cc CleanupContext
+	BeforeEach(func() {
+		createNamespace(nsLogicalVolumeTest)
+		cc = commonBeforeEach()
+	})
+	AfterEach(func() {
+		// When a test fails, I want to investigate the cause. So please don't remove the namespace!
+		if !CurrentSpecReport().State.Is(types.SpecStateFailureStates) {
+			kubectl("delete", "namespaces/"+nsLogicalVolumeTest)
+		}
+
+		commonAfterEach(cc)
+	})
+
+	k8sClient := createK8sClient()
+
+	It("should set Status.CurrentSize", func() {
+		pvcName := "check-current-size"
+		pvcYaml := fmt.Sprintf(pvcTemplateYAMLForLV, pvcName)
+		stdout, stderr, err := kubectlWithInput([]byte(pvcYaml), "apply", "-n", nsLogicalVolumeTest, "-f", "-")
+		Expect(err).ShouldNot(HaveOccurred(), "stdout=%s, stderr=%s", stdout, stderr)
+
+		By("checking that Status.CurrentSize exists")
+		var pvc corev1.PersistentVolumeClaim
+		Eventually(func(g Gomega) {
+			g.Expect(getObject(&pvc, "get", "pvc", "-n", nsLogicalVolumeTest, pvcName, "-ojson")).Should(Succeed())
```

Comment:
When we want to use `getObject`, we always want to do `kubectl get`.
Specifying "get" every time we use it seems to be redundant.

I would be happy if I could use it as follows (without "get").
```
getObject(&pvc, "pvc", "-n", nsLogicalVolumeTest, pvcName, "-ojson")
```

Reply:
Nice! I'll fix it.

Reply:
I also noticed that it needs `-ojson` always. I fixed it too.